geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/expression/FastDWithin.scala

```diff
@@ -41,14 +41,24 @@ object FastDWithin {
 
     override def evaluate(obj: AnyRef): Boolean = {
       val geom = exp1.evaluate(obj).asInstanceOf[Geometry]
-      if (geom == null || envelope.distance(geom.getEnvelopeInternal) > maxDegrees) { false } else {
```

Comment:
i'd like to keep this `envelope.distance(geom.getEnvelopeInternal) > maxDegrees` check, as it's very fast compared to the using the geodetic calculator. i think we can translate the envelope bounds in the same way we do for the geodetic calculations below, and if all of them are > maxDegrees we can return false directly.